src/WebJobs.Extensions.DocumentDB/Trigger/DocumentDBTriggerListener.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor;
+    using Microsoft.Azure.Documents.Client;
+    using Microsoft.Azure.WebJobs.Host.Executors;
+    using Microsoft.Azure.WebJobs.Host.Listeners;
+
+    internal class DocumentDBTriggerListener : IListener, IChangeFeedObserverFactory
+    {
+        private readonly ITriggeredFunctionExecutor executor;
+        private readonly ChangeFeedEventHost host;
+        public DocumentDBTriggerListener(ITriggeredFunctionExecutor executor, DocumentCollectionInfo documentCollectionLocation, DocumentCollectionInfo leaseCollectionLocation, ChangeFeedHostOptions leaseHostOptions)
+        {
+            this.executor = executor;
+            string hostName = Guid.NewGuid().ToString();
+            this.host = new ChangeFeedEventHost(hostName, documentCollectionLocation, leaseCollectionLocation, new ChangeFeedOptions(), leaseHostOptions);
+        }
+
+        public void Cancel()
+        {
+            this.StopAsync(CancellationToken.None).Wait();
+        }
+
+        public IChangeFeedObserver CreateObserver()
+        {
+            return new DocumentDBTriggerObserver(this.executor);
+        }
+
+        public void Dispose()
+        {
+            //Nothing to dispose
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            return host.RegisterObserverFactoryAsync(this);
```

Comment:
1. Partitions are handled dynamically, that's the reason for the Leases collection. If one instance of the Host is active, it will monitor all partitions, as more instances of the host are created (by multiplying instances in an App Service for example), partition ranges start to be distributed and leased. 
2. Yes, it's defined on the `leaseOptions` parameter of the constructor and exposed in the `DocumentDBConfiguration`. You can configure other things, like lease duration.
3. I'm not 100% sure of that scenario, from my tests, the processor waits until one batch is finished although there are configurable times in the `leaseOptions`. Since it's using a ContinuationToken to keep track of which was the last batch processed, you never lose changes if you are slow processing them, you just will get notified slower of the next batch.